### DISCUSSION TITLE
USB audio support?

### DISCUSSION THREAD

**Discussion by kaspernyhus** [pschatzmann/arduino-audio-tools#120] [Ideas]
Would USB audio be possible? esp32s2/3 has included an OTG feature.

Thanks,
Kasper Nyhus

**Comment by pschatzmann** [comment]
I think that would be a cool functionality and it should be possible to implement this. 
Unfortunately I don't have any of these processors...

**Reply by kaspernyhus** [reply]
Yes, very cool! The new esp32s3 is also very hard to come by these days.
We are a small team working on implementing it on said platform using the tinyusb stack, but are getting some errors unfortunately.
Hopefully we will be able to get some progress, and I would love to see it available here also.
All best,
Kasper Nyhus

**Comment by pschatzmann** [comment]
Hmm this is also something that should work on a RP2040...

**Reply by kaspernyhus** [reply]
I does! Apperently better than on the esp's atm as far as I have been able to confirm so far.
